pkg/archive/jar.go

```diff
@@ -170,3 +171,26 @@ func splitPreservingQuotes(s string, sep rune) []string {
 	}
 	return result
 }
+
+var StableGitProperties = ZipEntryStabilizer{
+	Name: "jar-git-properties",
+	Func: func(zf *MutableZipFile) {
+		// These files contain git properties set by git-commit-id-maven-plugin.
+		// They contain many unreproducible attributes as documented
+		// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/issues/825).
+		// Only JSON and properties file formats are available as documented
+		// [here](https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/95d616fc7e16018deff3f17e2d03a4b217e55294/src/main/java/pl/project13/maven/git/GitCommitIdMojo.java#L454).
+		// By default, these file are created in ${project.build.outputDirectory} and are hence at the root of the jar.
+		// We assume that the file name is 'git' as this is the default value for the plugin.
+		// We don't handle the case where the file name is changed by the user.
+		gitJsonRegex := regexp.MustCompile(`\bgit\.json$`)
+		if gitJsonRegex.MatchString(zf.Name) {
+			zf.SetContent([]byte("{}"))
+		}
+
+		gitPropertiesRegex := regexp.MustCompile(`\bgit\.properties$`)
```

Comment:
hmm this would match `foo-git.properties` so I think we'd want something more like `path.Base(zf.Name) == "git.properties"`.

same above with git.json.

Reply:
Yes, I agree. Good catch :)
Fixed in 0044c79

Reply:
Looks good!